File: backend/app/schemas/event.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class EventBase(BaseModel):
    name: str
    date: datetime
    venue_id: int
    event_type_id: int


class EventCreate(EventBase):
    pass


class EventUpdate(BaseModel):
    name: Optional[str] = None
    date: Optional[datetime] = None
    venue_id: Optional[int] = None
    event_type_id: Optional[int] = None


class EventResponse(EventBase):
    event_id: int
    organizer_id: int

    class Config:
        from_attributes = True


class EventTypeBase(BaseModel):
    name: str


class EventTypeCreate(EventTypeBase):
    pass


class EventTypeResponse(EventTypeBase):
    event_type_id: int

    class Config:
        from_attributes = True

File: frontend/src/types/event.ts
export interface EventType {
  event_type_id: number
  name: string
}

export interface EventTypeCreate {
  name: string
}

export interface EventBase {
  name: string
  date: string
  venue_id: number
  event_type_id: number
}

export interface EventCreate extends EventBase {}

export interface EventUpdate {
  name?: string
  date?: string
  venue_id?: number
  event_type_id?: number
}

export interface EventResponse extends EventBase {
  event_id: number
  organizer_id: number
}

export interface EventWithDetails extends EventResponse {
  venue_name?: string
  venue_city?: string
  event_type_name?: string
  organizer_name?: string
  available_tickets?: number
  total_tickets?: number
  min_price?: number
  max_price?: number
}
